flexget/plugins/list/radarr_list.py

```diff
@@ -0,0 +1,378 @@
+
+    def __len__(self):
+        return len(self.items)
+
+    def discard(self, value):
+        if not value:
+            return
+
+        if not isinstance(value, Entry):
+            raise TypeError()
+
+        matching_entry = self.__find_matching_entry(value)
+        if matching_entry:
+            movie_id = matching_entry['radarr_id']
+            response = self.service.delete_movie(movie_id)
+            log.verbose('Removed movie %s from Radarr', matching_entry['title'])
+            # Clear the cache
+            self._movie_entries = None
+        else:
+            log.debug('Could not find any matching movie to remove for entry %s', value)
+
+    def __ior__(self, other):
+        for entry in other:
+            if not isinstance(entry, Entry):
+                raise TypeError()
+
+            self.add(entry)
+
+    def __contains__(self, entry):
+        if not value:
+            return False
+
+        if not isinstance(value, Entry):
+            raise TypeError()
+
+        matching_entry = self.__find_matching_entry(value)
+
+        return not matching_entry is None
+
+    def add(self, entry):
+
+        if not isinstance(entry, Entry):
+            raise TypeError()
+
+        # The easiest way to add a movie to Radarr is to
+        # first use the lookup API. Using that we will get
+        # a json response which gives us most of the input
+        # we need for the PUT request.
+
+        result = None
+
+        # If the entry has a IMDB id, use that for lookup
+        if not result and 'imdb_id' in entry:
+            imdb_id = entry['imdb_id']
+            result = self.service.lookup_by_imdb(imdb_id)
+            if len(result) == 0:
+                result = None
+
+        # If the entry has a TMDB id, use that for lookup
+        if not result and 'tmdb_id' in entry:
+            tmdb_id = entry['tmdb_id']
+            result = self.service.lookup_by_tmdb(tmdb_id)
+            if len(result) == 0:
+                result = None
+
+        # Could not lookup by id. Try to use the title.
+        # However, we can only accept any results if it's
+        # one item, otherwise we don't know which to select.
+        if not result:
+            title = entry['title']
+            results = self.service.lookup_by_term(title)
+            if len(result) > 1:
+                log.debug(
+                    'The lookup for \'%s\' returned %d results. Using the first result \'%s\'.',
+                    title,
+                    len(results),
+                    results[0]['title']
+                )
+                result = results[0]
+            elif len(result) == 0:
+                result = None
+
+        if result:
+            rootfolders = self.service.get_rootfolders()
+            rootFolderPath = rootfolders[0]['path']
+
+            # TODO: should we let the user affect this one, 
+            # or try to parse the 'quality' value somehow?
+            qualityProfileId = 1
```

Comment:
This is if be for now. Let's discuss online how and if it should be handled differently